airflow/contrib/hooks/gcp_api_base_hook.py

```diff
@@ -131,7 +131,7 @@ def _get_access_token(self):
         """
         return self._get_credentials().token
 
-    def _authorize(self):
+    def authorize(self):
```

Comment:
Why do you think that this method should be public?  It seems to me that this should not be part of the public interface. This is the implementation detail for spefic library. It's should be protected.

Reply:
The problem with functions starting with `_` is that these can not be mocked by `unittest.mock` / `mock` because they are not public. They are kinda protected or private not intended to be tested.

Reply:
I am fine with protected. I didn't know that we use a design for protected members here.

To me private was always with `_` and public / protected without.

Reply:
@mik-laj You can add these design definition to your document I guess, because the python language has no such definition by default. It changes between projects.

Reply:
The protected/private is only a convention. I didn't think anything in checks if the method starts with `_` does it?

`_foo` as a method to me as a python developer says "Don't call this function form outside the class and have an easy ride when you upgrade"

Reply:
I checked it. Actually you are right. It works also with `_` functions. Don't know why I did that then.

Reply:
I was pretty confident it wasn't working :D

Reply:
I reverted the change to the `authorize` function.